backend/src/atlas_assistant/settings.py

```diff
@@ -1,17 +1,31 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
 from functools import lru_cache
 from pathlib import Path
-from typing import Literal
+from typing import Literal, TypeVar, override
 
 from langchain_chroma import Chroma
+from langchain_core.language_models.chat_models import BaseChatModel
 from langchain_mistralai import ChatMistralAI, MistralAIEmbeddings
-from pydantic import SecretStr
+from mistralai import Mistral
+from pydantic import BaseModel, Field, SecretStr
 from pydantic_settings import BaseSettings, SettingsConfigDict
 
+PydanticModel = TypeVar("PydanticModel", bound=BaseModel)
+
+Message = dict[str, str]
+
+
+class MistralConfig(BaseModel):
+    type: Literal["mistral"] = "mistral"
+    api_key: SecretStr | None = None
+    size: Literal["large"] | Literal["medium"] | Literal["small"] = "small"
+    temperature: float = 0.0
+
 
 class Settings(BaseSettings):
-    mistral_api_key: SecretStr | None = None
-    chat_model_size: Literal["large"] | Literal["medium"] | Literal["small"] = "small"
-    chat_model_temperature: float = 0.0
+    chat_model: MistralConfig = Field(discriminator="type")
```

Comment:
This `discriminator` lets us explicitly pick which type of model we want by setting `CHAT_MODEL__TYPE`, rather than relying on "smart" unions which can be a little fragile and magical.